### ISSUE TITLE
Too many automatic redirections were attempted.

### ISSUE THREAD

**Issue by Natfan** [JoeGandy/ShareX-Custom-Upload#7]
Hey all,

I'm trying to get this set up on my webserver. I copied the /src/ folder to /i/ (for images) and moved the uploader to my root http directory (/srv/http/) but I'm gettong the following error:

```
Error: Message:
Too many automatic redirections were attempted.

Request URL:
http://natfan.io/upload.php

Stack trace:
   at System.Net.HttpWebRequest.GetResponse()
   at ShareX.UploadersLib.Uploader.SendRequestFile(String url, Stream data, String fileName, String fileFormName, Dictionary`2 args, NameValueCollection headers, CookieCollection cookies, ResponseType responseType, HttpMethod method, String contentType, String metadata)
```

I've tried both http and https, along with natfan.io and www.natfan.io, but to no avail. The following is my config file, with the secure key redacted:

```
<?php

return array(
        /* This is a secure key that only you should know, an added layer of security for the ima
ge upload */
    'secure_key' => 'SECUREKEYISSUPERSECURE!',

    /* This is the url your output will be, usually http://www.domain.com/u/, also going to this
url will be the gallery page */
    'output_url' => 'http://natfan.io/i/',

    /* This is a redirect url if the script is accessed directly */
    'redirect_url' => 'http://natfan.io/',

    /* This is a list of IPs that can access the gallery page (Leave empty for universal access)
*/
    'allowed_ips' => array('192.168.0.0', '0.0.0.0'),

    /* Page title of the gallery page */
    'page_title' => 'Natfan\'s Image Uploader',

    /* Heading text at the top of the gallery page */
    'heading_text' => 'Natfan\'s Image Uploader',
);
```

[The following is a screenshot of my ShareX settings; again with the secure key redacted.](https://i.imgur.com/x6ExHmj.png)

Am I doing anything wrong or just being silly? I'm running Arch Linux with apache2.

**Comment by iamhansen** [comment]
the php file isnt closed 
```php
<?php

return array(
        /* This is a secure key that only you should know, an added layer of security for the ima
ge upload */
    'secure_key' => 'SECUREKEYISSUPERSECURE!',

    /* This is the url your output will be, usually http://www.domain.com/u/, also going to this
url will be the gallery page */
    'output_url' => 'http://natfan.io/i/',

    /* This is a redirect url if the script is accessed directly */
    'redirect_url' => 'http://natfan.io/',

    /* This is a list of IPs that can access the gallery page (Leave empty for universal access)
*/
    'allowed_ips' => array('192.168.0.0', '0.0.0.0'),

    /* Page title of the gallery page */
    'page_title' => 'Natfan\'s Image Uploader',

    /* Heading text at the top of the gallery page */
    'heading_text' => 'Natfan\'s Image Uploader',
);
?>
```
try this

**Comment by Natfan** [comment]
@iamhansen Sorry for the delay, now it seems to just output the HTML of my front page. I edited the `upload.php` file to use `i/config.php` and `/i/` in it's variables, rather than the defaulted `/u/`.

I also did what you recommended, silly me. I'm trying to learn PHP, should have seen such an obvious mistake!

**Comment by JoeGandy** [comment]
You shouldn't need to close a php file, so don't worry.

I'm worried you haven't uploaded everything to the right place?

If you are getting redirect back to your front page, then whats likely happening is you are getting sent back to your 'redirect_url' that you set in config.php 

`/* This is a redirect url if the script is accessed directly */
    'redirect_url' => 'http://natfan.io/',`

This only happens in two instances:

Your key is invalid, ensure that whatever you have set on this line
`/* This is a secure key that only you should know, an added layer of security for the ima
ge upload */
    'secure_key' => 'SECUREKEYISSUPERSECURE!',`
is correct and is also set in your shareX configuration like so:

http://jiy.io/18.05.37-07.11.17.png

I hope this is clear, if not have another read of the setup steps, as it should be correct, and if you are changing the path from 'u' to 'i' make sure its done in all the instances it occurs in upload.php

**Comment by Natfan** [comment]
I am a MORON!

I changed the upload.php to output both what the key in the config is showing and the key from the POST value. What I saw was that the config key was fine, but the POST key (sharex side of the config) was showing blank.

It turns out I had a space (" ") after the "key" variable, meaning it had the value "key ". Of course, this caused it to fail! Thank you so much for your help @JoeGandy!
